pkg/util/errors.go

```diff
@@ -149,6 +158,12 @@ func HandleStorageError(err error, entityName string) error {
 	default:
 		// in case we did not replace the pg.Error in the DB layer, propagate it as response message to give the caller relevant info
 		switch e := err.(type) {
+		case *ErrExistingReferenceEntityStorage:
+			return &HTTPError{
+				ErrorType:   "ExistingReferenceEntity",
+				Description: fmt.Sprintf("Error deleting entity %s: %s", entityName, err.Error()),
```

Comment:
This message behaviour is intentional.

Yes - it seems redundant when you look at the case when you try to delete a platform, but can't because of an existing instance: 

> "Error deleting entity **Platform**: Could not delete entity **Platform** due to existing reference entity **ServiceInstance**"

But if you try to delete a Broker, this message will look like this:

> "Error deleting entity **Broker**: Could not delete entity **ServicePlan** due to existing reference entity **ServiceInstance**"

So in cases where there are entities between the original entity and violation (reference) entity, like it is here with the **ServicePlan** entity, it's a much more useful message, as you get a bigger picture of what's going on. If I were to remove the part that you request, the message will even be strange in the case I want to delete a **Broker** with existing instances:

> "Could not delete entity **ServicePlan** due to existing reference entity **ServiceInstance**"